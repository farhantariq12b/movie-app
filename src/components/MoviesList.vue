<template>
  <div id="movie-list">
    <div v-if="loading" >Loading....</div>
    <div v-else class="flex flex-wrap justify-center" >
      <div data-testid="movie-list" v-for="(movie, index) in movies" :key="index">
        <MovieCard @toggleFavourite="toggleFavourite" :movie="movie"/>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { defineComponent, PropType } from 'vue';
import Movie from '@/types/Movie';
import MovieCard from '@/components/MovieCard.vue'

export default defineComponent({
  name: 'MovieList',
  components: {
    MovieCard
  },
  props: {
    movies: {
      required: true,
      type: Array as PropType<Movie[]>
    },
    loading: {
      type: Boolean,
    }
  },
  setup (props, { emit }) {
    function toggleFavourite (imdbID: string) {
      emit('toggleFavourite', imdbID)
    }
    return {
      toggleFavourite
    }
  }
});
</script>

<style scoped lang="scss">
</style>
