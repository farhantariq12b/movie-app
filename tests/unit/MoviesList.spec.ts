import { shallowMount } from "@vue/test-utils";
import MoviesList from "@/components/MoviesList.vue";

describe("MovieCard.vue", () => {
  it("should render HTML correctly", () => {
    const movies = [
      {
        Title: 'Water',
        Year: '2021',
        imdbID: '1234',
        favourite: true
      }
    ];
    const wrapper = shallowMount(MoviesList, {
      props: { movies },
    });
		expect(wrapper.vm.$el).toMatchSnapshot();
	});

  it("renders movies props correctly when passed", () => {
    const movies = [
      {
        Title: 'Water',
        Year: '2021',
        imdbID: '1234',
        favourite: true
      },
      {
        Title: 'Sea',
        Year: '2021',
        imdbID: '1235',
        favourite: false
      }
    ];
    const wrapper = shallowMount(MoviesList, {
      props: { movies, loading: false },
    });
		expect(wrapper.vm.$props.movies.length).toBe(2)
  });

});
