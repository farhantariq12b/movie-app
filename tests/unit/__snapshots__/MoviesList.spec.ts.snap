// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard.vue should render HTML correctly 1`] = `
<div
  id="movie-list"
>
  <div
    class="flex flex-wrap justify-center"
  >
    
    <div
      data-testid="movie-list"
    >
      <movie-card-stub
        movie="[object Object]"
      />
    </div>
    
  </div>
</div>
`;
